examples/chat.rs

```diff
@@ -127,7 +124,7 @@ struct Peer {
     ///
     /// This is used to receive messages from peers. When a message is received
     /// off of this `Rx`, it will be written to the socket.
-    rx: Rx,
+    rx: Pin<Box<dyn Stream<Item = String> + Send>>,
```

Comment:
This raises a question. The `Rx` type is not used in a way that expects a `Stream`, rather just in a way that expects something with a `poll_next` method. Currently the `impl Stream` is just removed, and the private `poll_recv` method is not made public.

What approach should we be taking here? I kinda feel like we should be exposing the `poll_recv` method, as the type supports it perfectly fine (there is only one receiver, so there are no intrusiveness concerns), but then we have both `poll_next` and `poll_recv` when we add back the `Stream` impl in ≥ half a year.

Reply:
Does this example need to use `poll_*` APIs at all?

Reply:
Yeah, I think we should rewrite it.